### PR TITLE
ARM64 support (ABC-436)

### PR DESCRIPTION
**Pull request by thomas-tu**
## Purpose of this PR

**Ticket/Jira #:** https://jira.unity3d.com/browse/ABC-436

Changes:
- Depends on ISPC 1.21.0 (was 1.14.1).
- Depends on Alembic 1.8.5 (was 1.7.16).
- Small change on internal API (was crashing on ARM64 platform).
- Updated cmake files to generate ARM64 .dll. There's now a new .meta file in the package Plugins/ folder with specific import settings for the new .dll.

Note: at the moment, ARM64 .dll is manually built until we can make it automatic in the build pipeline.

## Testing

**Functional Testing status:**

- Manually tested on ARM64 platform.
- Usually CI tests.

**Performance Testing status:**

## Overall Product Risks

**Complexity:** Low

**Halo Effect:** Medium

## Documentation & UX Writing

| User facing text to review |  | Details |
| :--- | :--- | :--- |
| User interface | N |  |
| Public API docs | N |  |
| Changelog | Y |  |

| Documentation halo effect |  | Jira link |
| :--- | :--- | :--- |
| User manual | N |  |
| Other docs | N |  |

## Additional information

**Note to reviewers:**

**Reminder:**

- [x] Add entry in CHANGELOG.md

### CODE REVIEW

**Review by windxu88 — APPROVED** [review]
Ran nightly trigger and all tests passed. Left one small question in comment.

File: Source/abci/CMakeLists.txt [hunk 1]
```diff
@@ -16,10 +16,17 @@ file(GLOB sources *.cpp *.h Foundation/*.h Foundation/*.cpp Importer/*.h Importe
 list(APPEND sources ${ISPC_SOURCES})
 list(APPEND sources ${ISPC_OBJECTS})
 
-add_plugin(abci
-    SOURCES ${sources}
-    PLUGINS_DIR ${CMAKE_INSTALL_PREFIX}/com.unity.formats.alembic/Runtime/Plugins/x86_64
+if (CMAKE_SYSTEM_PROCESSOR STREQUAL "ARM64")
```
Comment (windxu88) [inline Source/abci/CMakeLists.txt:19]: I am curious what will happen if building on a Silicon Mac? Will the Mac lib also end up in `Plugins/ARM64` folder?

Comment (thomas-tu) [reply Source/abci/CMakeLists.txt:19]: Very good point, that's is possibly correct.

File: com.unity.formats.alembic/CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [2.4.0-pre.1] - 2023-11-07
 
+### Added
+- Add library for Windows ARM64 platform.
```
Comment (arejayelle) [inline com.unity.formats.alembic/CHANGELOG.md:10]: wording here is a bit weird. 
```suggestion
- Add library for Windows ARM64 platform support.
```

**Review by arejayelle — APPROVED** [review]
the code logic seems fine to me. Can't test it myself though

**Review by SanaJal — APPROVED** [review]
LGTM !

**Review by cguer — APPROVED** [review]
QA approved 👍!

Did some basic testing and didn't find any obvious issues although I will only be 100% confident once the CI is running.

Tested on:
- 2023.3.0a16 (Windows Arm64)
- 2023.3.0a16 (MacOS Arm64/Silicon)

**Pull request merged**